src/network/addrthread.py

```diff
@@ -15,15 +22,26 @@ def run(self):
             while True:
                 try:
                     data = addrQueue.get(False)
-                    chunk.append((data[0], data[1]))
-                    if len(data) > 2:
-                        source = BMConnectionPool().getConnectionByAddr(data[2])
+                    chunk.append(data)
                 except Queue.Empty:
                     break
                 except KeyError:
                     continue
 
-            # finish
+            # Choose peers randomly
```

Comment:
Shouldn't this block be inside the loop?

Reply:
I don't think so, it works the same way as `invthread.py`

Reply:
maybe add
```python
if not chunk:
    continue
```

Reply:
it eats 99% CPU

Reply:
yea I just noticed that too